packages/typescript-eslint/src/config-helper.ts

```diff
@@ -85,11 +85,30 @@ export interface ConfigWithExtends extends TSESLint.FlatConfig.Config {
 export function config(
   ...configs: ConfigWithExtends[]
 ): TSESLint.FlatConfig.ConfigArray {
-  return configs.flatMap(configWithExtends => {
+  return configs.flatMap((configWithExtends, configIndex) => {
     const { extends: extendsArr, ...config } = configWithExtends;
     if (extendsArr == null || extendsArr.length === 0) {
       return config;
     }
+    const undefinedExtensions = extendsArr.reduce<number[]>(
+      (acc, extension, extensionIndex) => {
+        const maybeExtension = extension as
+          | TSESLint.FlatConfig.Config
+          | undefined;
+        if (!maybeExtension) {
+          acc.push(extensionIndex);
+        }
+        return acc;
+      },
+      [],
+    );
+    if (undefinedExtensions.length) {
+      throw new Error(
+        `Your config at index ${configIndex} contains undefined extensions ` +
```

Comment:
IMO we shouldn't say it's `undefined` unless we specifically check for `undefined` above. As written, the invalid extension could be any other falsey value that's equally disallowed by the declared type of the `extends` property. Let's synchronize the check and the wording? (no opinion on which one changes, or both).

For instance, right now, you have the following behavior, which is misleading
```ts
    expect(() =>
      plugin.config(
        {
          extends: [extension],
          files: ['common-file'],
          ignores: ['common-ignored'],
          name: 'my-config',
          rules: { rule: 'error' },
        },
        {
          extends: [
            // eslint-disable-next-line @typescript-eslint/no-explicit-any
            undefined as any,
            extension,
            null!,
            // eslint-disable-next-line @typescript-eslint/no-explicit-any
            false as any,
            // eslint-disable-next-line @typescript-eslint/no-explicit-any
            0 as any,
            // eslint-disable-next-line @typescript-eslint/no-explicit-any
            '' as any,
          ],
          files: ['common-file'],
          ignores: ['common-ignored'],
          name: 'my-config',
          rules: { rule: 'error' },
        },
      ),
    ).toThrow(
      'Your config at index 1 contains undefined extensions at the following' +
        ' indices: 0, 2, 3, 4, 5',
    );
```

WDYT?

Reply:
ok, i've changed to explicitly checking for null/undefined instead of any falsey value. ofc, if someone puts nonsense into their `extends` array, they'll be back to getting a cryptic error but on a practical level, we cannot runtime check every mistake a user might make and checking for undefined will cover 99.99% of the errors that i'd expect here.

Reply:
or said another way: my goal with this PR is to catch the obvious error that a lot of folks are getting upgrading to 8.10.0. it seems like we never really did much runtime checking anywhere else either.